backend/generator.js

```diff
@@ -41,10 +39,42 @@ exports.executeScript = function (socket, formData) {
   process.on('exit', function (code) {
     console.log('child process exited with code ' + code);
     if (code === 0) {
-      socket.emit('success', {email: email, uniqueId: uniqueId});
+      socket.emit('success', {email: email, uniqueId: uniqueId, gitUrl: gitUrl});
     } else {
       socket.emit('failure', {errorCode: code});
     }
   });
   return true;
-};
\ No newline at end of file
+};
+
+exports.deployPages = function (socket, data) {
+  var donePercent = 0;
+  var projectName = data.gitURL.split("/")[4].split(".")[0];
+  var gitURL = "http://" + data.username+":" + crypter.decrypt(data.encryptedToken) + "@github.com/" + data.username + "/" + projectName;
```

Comment:
Instead of building the url here, you could have passed the token as OAUTH_TOKEN and the publish script should have handled that.